### DISCUSSION TITLE
Is it possible to secure the logs in k8s?

### DISCUSSION THREAD

**Discussion by chriskeAlerant** [apache/airflow#19243] [Q&A] [answered]
Hi there,

I'm running Airflow in an OpenShift environment with k8sExecutor. I have a PVC for logs and it is mounted to every pod, including the workers. Everything works fine. Now, we want to use our instance in a multi-tenant way, which means it is insecure if any DAG can read every other DAG's log, because every pod has the common log PVC attached. Is there any way to restrict that access for the DAGs? What do you think?

Thanks in advance

**Comment by potiuk** [comment, marked as answer]
Well. If you want to use mutli-tenant for single Airflow instance, then it's already pretty bad. Because every DAG can not only read logs but also READ and WRITE(!) all the DB for all the other DAGs. Reading logs is your least worry in this case I think.

Unless you want to have separate instances of Airflow and in this case you can simply have separate log storage for each instance.

Lack of multi-tenancy  is a known limitation of Airflow - it cannot (yet) be implemented as multi-tennant, because of this very reason.It's coming in 2.3 most likely - and the most you can do now is to implement separate airflow instance for every tenant. You can reuse some resources - you could use same db server (with separate schema for each tenant) and simply run your airflows on the same clusters (each tenant in different namespace) and share authentication integration with your enterprise authentication system (for example using keycloak). But it's impossible to have secure multi-tenant single-Airflow installation yet.

**Comment by chriskeAlerant** [comment]
Thank you for the explanation!
